packages/ag-charts-community/src/chart/chart.ts

```diff
@@ -1369,13 +1351,20 @@ export abstract class Chart extends Observable implements AgChartInstance {
     async waitForUpdate(timeoutMs = 5000): Promise<void> {
         const start = performance.now();
 
-        while (this._pendingFactoryUpdates.length > 0 || this.updatePending) {
+        if (this._pendingFactoryUpdatesCount > 0) {
+            // Await until any pending updates are flushed through.
+            await this.updateMutex.acquire(async () => undefined);
+        }
```

Comment:
Is `_pendingFactoryUpdatesCount` needed if we are waiting for updates to be flushed anyway?

Also, is it worth making `.acquire(async () => undefined)` it's own function on `Mutex`, seems like it may be useful in situations opposite to `.acquireImmediately()`? Like perhaps `.acquireWhenReady()` or some better name.

Reply:
So I tried to remove `_pendingFactoryUpdateCount`, but unfortunately since there is a gap between updates being applied and the `performUpdate()` cycle executing, this was the only reliable way to ensure that all applied updates have been applied AND then that the resulting `performUpdate()` has happened.

Reply:
Added `Mutex.waitForClearAcquireQueue()` for this case as suggested.